nvflare/private/fed/server/server_command_agent.py

```diff
@@ -12,60 +12,101 @@
+from nvflare.private.defs import CellChannel, CellMessageHeaderKeys, new_cell_message
 
 from .server_commands import ServerCommands
 
 
 class ServerCommandAgent(object):
-    def __init__(self, listen_port) -> None:
+    def __init__(self, engine, cell: Cell) -> None:
         """To init the CommandAgent.
 
         Args:
             listen_port: port to listen the command
         """
         self.logger = logging.getLogger(self.__class__.__name__)
-        self.listen_port = int(listen_port)
-        self.thread = None
+        # self.listen_port = int(listen_port)
+        # self.thread = None
         self.asked_to_stop = False
+        self.engine = engine
+        self.cell = cell
 
-    def start(self, engine):
-        self.thread = threading.Thread(
-            target=listen_command, args=[self.listen_port, engine, self.execute_command, self.logger]
+    def start(self):
+        self.cell.register_request_cb(
+            channel=CellChannel.SERVER_COMMAND,
+            topic="*",
+            cb=self.execute_command,
         )
-        self.thread.start()
-        self.logger.info(f"ServerCommandAgent listening on port: {self.listen_port}")
-
-    def execute_command(self, conn, engine):
-        while not self.asked_to_stop:
-            try:
-                if conn.poll(1.0):
-                    msg = conn.recv()
-                    msg = fobs.loads(msg)
-                    command_name = msg.get(ServerCommandKey.COMMAND)
-                    data = msg.get(ServerCommandKey.DATA)
-                    command = ServerCommands.get_command(command_name)
-                    if command:
-                        with engine.new_context() as new_fl_ctx:
-                            reply = command.process(data=data, fl_ctx=new_fl_ctx)
-                            if reply is not None:
-                                conn.send(reply)
-            except EOFError:
-                self.logger.info("listener communication terminated.")
-                break
-            except Exception as e:
-                self.logger.error(
-                    f"IPC Communication error on the port: {self.listen_port}: {secure_format_exception(e)}."
-                )
+        self.cell.register_request_cb(
+            channel=CellChannel.AUX_COMMUNICATION,
+            topic="*",
+            cb=self.aux_communicate,
+        )
+        self.logger.info(f"ServerCommandAgent cell start: {self.cell.get_fqcn()}")
+
+    def execute_command(self, request: CellMessage) -> CellMessage:
+
+        assert isinstance(request, CellMessage), "request must be CellMessage but got {}".format(type(request))
+
+        command_name = request.get_header(MessageHeaderKey.TOPIC)
+        data = fobs.loads(request.payload)
+
+        token = request.get_header(CellMessageHeaderKeys.TOKEN, None)
+        client = None
+        if token:
+            client = self.engine.server.client_manager.clients.get(token)
```

Comment:
It feels weird to accessing the instance variable like this, we should introduce some getting methods?

Reply:
wrapped in _get_client()